api/v1alpha4/conversion.go

```diff
@@ -16,12 +16,161 @@ limitations under the License.
 
 package v1alpha4
 
-func (*Metal3Cluster) Hub()             {}
-func (*Metal3ClusterList) Hub()         {}
-func (*Metal3Machine) Hub()             {}
-func (*Metal3MachineList) Hub()         {}
-func (*Metal3MachineTemplate) Hub()     {}
-func (*Metal3MachineTemplateList) Hub() {}
-func (*Metal3DataTemplate) Hub()        {}
-func (*Metal3Data) Hub()                {}
-func (*Metal3DataClaim) Hub()           {}
+import (
+	"github.com/metal3-io/cluster-api-provider-metal3/api/v1alpha5"
+	apiconversion "k8s.io/apimachinery/pkg/conversion"
+	utilconversion "sigs.k8s.io/cluster-api/util/conversion"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
+
+//Constant variables
+const (
+	APIEndpointPort = "6443"
+)
+
+func (src *Metal3Cluster) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1alpha5.Metal3Cluster)
+	return Convert_v1alpha4_Metal3Cluster_To_v1alpha5_Metal3Cluster(src, dst, nil)
+}
+
+func (dst *Metal3Cluster) ConvertFrom(srcRaw conversion.Hub) error {
+	src := srcRaw.(*v1alpha5.Metal3Cluster)
+	return Convert_v1alpha5_Metal3Cluster_To_v1alpha4_Metal3Cluster(src, dst, nil)
+}
+
+func (src *Metal3ClusterList) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1alpha5.Metal3ClusterList)
+	return Convert_v1alpha4_Metal3ClusterList_To_v1alpha5_Metal3ClusterList(src, dst, nil)
+}
+
+func (dst *Metal3ClusterList) ConvertFrom(srcRaw conversion.Hub) error {
+	src := srcRaw.(*v1alpha5.Metal3ClusterList)
+	return Convert_v1alpha5_Metal3ClusterList_To_v1alpha4_Metal3ClusterList(src, dst, nil)
+}
+
+func (src *Metal3Machine) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1alpha5.Metal3Machine)
+	if err := Convert_v1alpha4_Metal3Machine_To_v1alpha5_Metal3Machine(src, dst, nil); err != nil {
+		return err
+	}
+
+	// Manually restore data.
+	restored := &v1alpha5.Metal3Machine{}
+	if ok, err := utilconversion.UnmarshalData(src, restored); err != nil || !ok {
+		return err
+	}
+
+	dst.Spec.MetaData = restored.Spec.MetaData
+	dst.Spec.NetworkData = restored.Spec.NetworkData
+	dst.Spec.DataTemplate = restored.Spec.DataTemplate
+	dst.Spec.Image = restored.Spec.Image
+	dst.Status.UserData = restored.Status.UserData
+	dst.Status.MetaData = restored.Status.MetaData
+	dst.Status.NetworkData = restored.Status.NetworkData
+	dst.Status.RenderedData = restored.Status.RenderedData
```

Comment:
currently v1a4 and v1a5 are the same so you should not need this.

Reply:
Removed